src/reference/asciidoc/testing.adoc

```diff
@@ -2,12 +2,278 @@
+handler.handleMessage(message);
+verify(handler).handleMessage(messageWithPayload(SOME_PAYLOAD));
+verify(handler).handleMessage(messageWithPayload(is(instanceOf(Date.class))));
+...
+MessageChannel channel = mock(MessageChannel.class);
+when(channel.send(messageWithHeaderEntry(SOME_HEADER_KEY, is(instanceOf(Short.class)))))
+        .thenReturn(true);
+assertThat(channel.send(message), is(false));
+----
+
+More utils can be added or migrated eventually.
+For example `RemoteFileTestSupport` implementations for the (S)FTP tests can be moved from the `test` directory of the particular modules to this publicly available `spring-integration-test-support` artifact.
+
+[[test-context]]
+=== Spring Integration and test context
+
+Typically tests for Spring applications use the Spring Test Framework and since Spring Integration is fully based on the Spring Framework foundation, everything what we can do with the Spring Test Framework is applied as well when we come to the task for testing integration flows.
+The `org.springframework.integration.test.context` package provides some components for enhancing test context for integration needs.
+First of all we should configure our test class with particular `@SpringIntegrationTest` annotation to enable Spring Integration Test Framework:
+
+[source,java]
+----
+@RunWith(SpringRunner.class)
+@SpringIntegrationTest(noAutoStartup = {"inboundChannelAdapter", "*Source*"})
+public class MyIntegrationTests {
+
+    @Autowired
+    private MockIntegrationContext mockIntegrationContext;
+
+}
+----
+
+The `@SpringIntegrationTest` populates `MockIntegrationContext` bean which can be autowired to the test class for further usage.
+With the provided `noAutoStartup` option, the Spring Integration Test Framework switches off endpoints, which are matched to the provided patterns, from the auto-startup during application context refreshing.
+This is useful, when we would like do not have real connections to the target systems from Inbound Channel Adapters, for example AMQP Inbound Gateway, JDBC Polling Channel Adapter, WebSocket Message Producer in client mode etc.
+
+The `MockIntegrationContext` is aimed to be used in the target test-cases for modifications beans in the real application context, for example those stopped endpoints can be replaced with mocks:
+
+[source,java]
+----
+@Test
+public void testMockMessageSource() {
+    MessageSource<String> messageSource = () -> new GenericMessage<>("foo");
+
+    this.mockIntegrationContext.instead("mySourceEndpoint", messageSource);
+
+    Message<?> receive = this.results.receive(10_000);
+    assertNotNull(receive);
+}
+----
+
+See their JavaDocs for more information.
+
+[[testing-mocks]]
+=== Integration Mocks
+
+The `org.springframework.integration.test.mock` package offers tools and utilities for mocking, stubbing and verifications of Spring Integration components.
+The mocking functionality is fully based and compatible with well know Mockito Framework.
```

Comment:
Probably it's worth to explicitly denote the version of Mockito (2) because there are a lot of Mockito 1 users (for example those who have to use its PowerMockito extension which is not yet fully compatible with version 2).